decompiler/frontend/binaryninja/handlers/assignments.py

```diff
@@ -0,0 +1,176 @@
+
+        In case higher registers use masking
+        e.g. eax.ah = x <=> eax = (eax & 0xffff00ff) + (x << 2)
+        """
+        if assignment.offset == 0:
+            destination = self._lift_contraction(assignment, is_aliased=is_aliased)
+            value = self._lifter.lift(assignment.src)
+        else:
+            destination = self._lifter.lift(assignment.dest, is_aliased=is_aliased, parent=assignment)
+            value = self._lift_masked_assignment(assignment)
+        return Assignment(destination, value)
+
+    def lift_get_field(self, instruction: mediumlevelil.MediumLevelILVarField, is_aliased=False, **kwargs) -> Operation:
+        """
+        Lift an instruction accessing a field from the outside.
+        e.g. x = eax.ah <=> x = eax & 0x0000ff00
+        """
+        source = self._lifter.lift(instruction.src, is_aliased=is_aliased, parent=instruction)
+        cast_type = source.type.resize(instruction.size * self.BYTE_SIZE)
+        if instruction.offset:
+            return BinaryOperation(
+                OperationType.bitwise_and,
+                [source, Constant(self._get_all_ones_mask_for_type(instruction.size) << instruction.offset)],
+                vartype=cast_type,
+            )
+        return UnaryOperation(OperationType.cast, [source], vartype=cast_type, contraction=True)
+
+    def lift_store(self, assignment: mediumlevelil.MediumLevelILStoreSsa, **kwargs) -> Assignment:
+        return Assignment(
+            UnaryOperation(
+                OperationType.dereference,
+                [op := self._lifter.lift(assignment.dest, parent=assignment)],
+                vartype=op.type,
+                writes_memory=assignment.dest_memory,
+            ),
+            self._lifter.lift(assignment.src),
+        )
+
+    def _lift_contraction(self, assignment: mediumlevelil.MediumLevelILSetVarField, is_aliased=False, **kwargs) -> UnaryOperation:
+        """
+        We lift assignment to lower register part (offset 0 from register start) as contraction (cast)
+
+        E.g.:
+        eax.al = 10;
+        becomes:
+        (byte) eax = 10; // Assign(Cast([eax], byte, contraction=true), Constant(10))
+        :param instruction: instruction of type MLIL_SET_VAR_FIELD
+        """
+        destination_operand = self._lifter.lift(assignment.dest, is_aliased=is_aliased, parent=SetVar)
+        contraction_type = destination_operand.type.resize(assignment.size * self.BYTE_SIZE)
+        return UnaryOperation(OperationType.cast, [destination_operand], vartype=contraction_type, contraction=True)
+
+    def _lift_masked_assignment(self, assignment: mediumlevelil.MediumLevelILSetVarField, **kwargs) -> BinaryOperation:
+        return BinaryOperation(
+            OperationType.bitwise_or,
+            [
+                BinaryOperation(
+                    OperationType.bitwise_and,
+                    [
+                        self._lifter.lift(assignment.prev, parent=assignment),
```

Comment:
I think we need to further pass aliased info here.

Reply:
fixed in cb6e63d8